BricksAndHearts.UnitTests/ServiceTests/Landlord/LandlordServiceTests.cs

```diff
@@ -19,21 +21,278 @@ public LandlordServiceTests(TestDatabaseFixture fixture)
     {
         Fixture = fixture;
     }
+
+    //RegisterLandlord can't be tested as it uses a transaction
+
+    [Fact]
+    public void GetLandlordIfExistsFromId_ReturnsLandlord_WithUniqueId()
+    {
+        // Arrange
+        using var context = Fixture.CreateReadContext();
+        var service = new LandlordService(context);
+
+        // Act
+        var result = service.GetLandlordIfExistsFromId(4).Result;
+
+        // Assert
+        result.Should().BeOfType<LandlordDbModel>().Which.Id.Should().Be(4);
+    }
+
+    [Fact]
+    public void GetLandlordIfExistsFromId_ReturnsNull_WithInvalidId()
+    {
+        // Arrange
+        using var context = Fixture.CreateReadContext();
+        var service = new LandlordService(context);
+
+        // Act
+        var result = service.GetLandlordIfExistsFromId(1000).Result;
+        var result2 = service.GetLandlordIfExistsFromId(null).Result;
+
+        // Assert
+        result.Should().BeNull();
+        result2.Should().BeNull();
+    }
+
+    [Fact]
+    public void GetLandlordIfExistsWithProperties_ReturnsLandlord_WithUniqueId()
+    {
+        // Arrange
+        using var context = Fixture.CreateReadContext();
+        var service = new LandlordService(context);
+
+        // Act
+        var result = service.GetLandlordIfExistsWithProperties(1).Result;
+
+        // Assert
+        result.Should().BeOfType<LandlordDbModel>().Which.Id.Should().Be(1);

```

Comment:
I assumed this too, I actually don't think the method checks for this at all based on running the tests? Might just be me misinterpreting the results but if they had no properties it still seemed to return the landlord, just with an empty list of properties.

Reply:
I think that's correct, it just uses `.Include` to load the properties if they exist, otherwise you would have to do it in a separate database query.

I assume landlord number 1 does have some properties in our test setup, but I'm not sure

Reply:
I'll add some properties to landlord 1 if not and then check for non null / empty :)